### PR TITLE
Wolframite for scientists

### PR DESCRIPTION
**Pull request by light-matters**
Experimenting with creating a draft pull request. Something's not working here though. Cf. #76

### CODE REVIEW

**Comment by light-matters** [comment]
Got around #76 for now.

**Comment by holyjak** [comment]
Currently, building the site fails with

> Symbols passed to Mathematica must be alphanumeric (apart from forward slashes and dollar signs). Passed: ** Known aliases:

I assume this is b/c we run another, argument-less `wl/start` before the one in this ns. So either we need to support re-start / reset[-aliases], or you need to run stop & start, or you need to do st. like

```clj
(defn my-eval [expr] (wl/eval expr {:aliases aliases})
```

WDYT?

**Comment by holyjak** [comment]
BTW notice the https://github.com/scicloj/wolframite/pull/77/commits/a2f2de63575eb44dafe13f73e20d1ef3ddd3afd9 I added

**Comment by light-matters** [comment]
> Currently, building the site fails with
> 
> > Symbols passed to Mathematica must be alphanumeric (apart from forward slashes and dollar signs). Passed: ** Known aliases:
> 
> I assume this is b/c we run another, argument-less `wl/start` before the one in this ns. So either we need to support re-start / reset[-aliases], or you need to run stop & start, or you need to do st. like
> 
> ```clojure
> (defn my-eval [expr] (wl/eval expr {:aliases aliases})
> ```
> 
> WDYT?

Restart would be the cleanest thing here I think, but start and stop don't currently do what we want at the moment. If I really have to though then it's good to be reminded about just passing options to eval.

**Comment by holyjak** [comment]
Pls see if https://github.com/scicloj/wolframite/pull/88 helps with your alias/restart issues

**Comment by light-matters** [comment]
Changed start to restart. Good to go (for now) from my side. I need to learn how to make the book/website for proper testing though.

**Review by holyjak — CHANGES_REQUESTED** [review]
Some of symbols in wolframite.wolfram are defined twice (as clj-kondo warns me):

```clj
(def _> "Maps to the Wolfram function RuleDelayed" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote _>)))
(def <_> "Maps to the Wolfram function Dot" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote <_>)))

(def _> "Maps to the Wolfram function Replace" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote _>)))
(def <_> "Maps to the Wolfram function Expand" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote <_>)))
```

File: src/wolframite/wolfram.clj [hunk 1]
```diff
@@ -6320,29 +6320,28 @@
...
 (def || "Maps to the Wolfram function Or" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote ||)))
-(def ..= "Maps to the Wolfram function SetDelayed" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote ..=)))
 (def <= "Maps to the Wolfram function LessEqual" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote <=)))
 (def * "Maps to the Wolfram function Times" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote *)))
 (def -> "Maps to the Wolfram function Rule" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote ->)))
-(def fn "Maps to the Wolfram function Function" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote fn)))
 (def do "Maps to the Wolfram function CompoundExpression" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote do)))
 (def > "Maps to the Wolfram function Greater" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote >)))
+(def _> "Maps to the Wolfram function RuleDelayed" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote _>)))
+(def <_> "Maps to the Wolfram function Dot" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote <_>)))
```
Comment (holyjak) [inline src/wolframite/wolfram.clj:6334]: this should be `<*>`. How is this possible? Was this properly generated via `wolframite.impl.wolfram-syms.write-ns/write-ns!` ?!

File: src/wolframite/wolfram.clj [hunk 2]
```diff
@@ -6351,4 +6350,16 @@
...
   ;; we need the whole body to be pure symbols and primitives only, which Wolfram will understand
   (let [symbolic-sexp (clojure.walk/prewalk wolframite.impl.wolfram-syms.intern/try->wolf-sym sexp)]
-    `(list (quote ~'Function) (quote ~args) (quote ~symbolic-sexp))))
\ No newline at end of file
+    `(list (quote ~'Function) (quote ~args) (quote ~symbolic-sexp))))
+(def >_< "Maps to the Wolfram function Simplify" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote >_<)))
+(def ** "Maps to the Wolfram function Power" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote **)))
+(def >>_<< "Maps to the Wolfram function FullSimplify" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote >>_<<)))
+(def <<_>> "Maps to the Wolfram function ExpandAll" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote <<_>>)))
+(def _> "Maps to the Wolfram function Replace" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote _>)))
```
Comment (holyjak) [inline src/wolframite/wolfram.clj:6358]: Conflicts with RuleDelayed

File: src/wolframite/wolfram.clj [hunk 3]
```diff
@@ -6351,4 +6350,16 @@
   ;; If there is any w/ symbol such as w/Plus, replace it just with the fn name, `Plus` -
   ;; we need the whole body to be pure symbols and primitives only, which Wolfram will understand
   (let [symbolic-sexp (clojure.walk/prewalk wolframite.impl.wolfram-syms.intern/try->wolf-sym sexp)]
-    `(list (quote ~'Function) (quote ~args) (quote ~symbolic-sexp))))
\ No newline at end of file
+    `(list (quote ~'Function) (quote ~args) (quote ~symbolic-sexp))))
+(def >_< "Maps to the Wolfram function Simplify" (wolframite.impl.wolfram-syms.intern/wolfram-fn (quote >_<)))
```
Comment (holyjak) [inline src/wolframite/wolfram.clj:6354]: Where does this come from? Was it added manually, or...?

**Review by holyjak — CHANGES_REQUESTED** [review]

**Comment by holyjak** [comment]
BTW, how has the `wolframite.wolfram-extended` been generated?!

File: src/wolframite/wolfram.clj (hunk 3 above)
Comment (light-matters) [reply src/wolframite/wolfram.clj:6354]: This was the leftover from my first attempt at changing the symbol aliases. I realised afterwards that I'd overloaded some symbols though and changed it. Either I didn't rewrite wolfram... or I didn't commit the change.

Comment (light-matters) [reply src/wolframite/wolfram.clj:6354]: Mmm there's something else going on as changes I'm sure I made are not here. I wonder if updated the wrong branch? Anyways, I'll sort it out now.

**Comment by light-matters** [comment]
> BTW, how has the `wolframite.wolfram-extended` been generated?!
> 
> Ideally, its docstring should describe how it differs from ...wolfram.

It was generated in the same way as wolfram.clj. As such, the namespace and, now, docstring had to be changed manually. This is something that I would like to develop more in the future. The idea being that wolfram.clj will be the most conventional, least opinionated mapping to Wolfram functions, whereas wolfram_extended will gradually develop to include any UX convenience that 'helps' with working with Wolfram, whether it fits existing Wolfram conventions or not.

File: src/wolframite/wolfram.clj (hunk 2 above)
Comment (light-matters) [reply src/wolframite/wolfram.clj:6358]: fixed

File: src/wolframite/wolfram.clj (hunk 1 above)
Comment (light-matters) [reply src/wolframite/wolfram.clj:6334]: it was generated with write-ns, but it was a previous idea for the Dot symbol

File: notebooks/for_scientists/index.clj [hunk 4]
```diff
@@ -14,7 +14,7 @@
   ;; - are clay kinds executed outside of a clay environment, i.e. are they treated like 'comment's under 'normal' circumstances?
   ;; - Move general notes and examples out from the Cavity section
   ;; - should probably be split into two (the cavity part can be quite self-contained)
-  ,)
```
Comment (holyjak) [inline notebooks/for_scientists/index.clj]: the comma was there on purpose, so that parinfer does not move the closing paren :-)

**Review by holyjak — APPROVED** [review]

File: notebooks/for_scientists/index.clj [hunk 5]
```diff
@@ -14,7 +14,7 @@
...
+  )
```
Comment (holyjak) [inline notebooks/for_scientists/index.clj:17]:
```suggestion
  ,)
```
Keep the comma, to prevent parinfer from moving the paren.

**Pull request merged**